lms/djangoapps/teams/tests/test_views.py

```diff
@@ -523,3 +568,224 @@ def test_invalid_course_id(self, course_id):
 
     def test_invalid_topic_id(self):
         self.get_topic_detail('foobar', self.test_course_1.id, 404)
+
+
+@ddt.ddt
+class TestListMembershipAPI(TeamAPITestCase):
+    """Test cases for the membership list endpoint."""
+
+    @ddt.data(
+        (None, 403),
+        ('student_inactive', 403),
+        ('student_unenrolled', 404),
+        ('student_enrolled', 200),
+        ('student_enrolled_both_courses_other_team', 200),
+        ('staff', 200),
+        ('course_staff', 200),
+    )
+    @ddt.unpack
+    def test_access(self, user, status):
+        membership = self.get_membership_list(status, {'team_id': self.test_team_1.team_id}, user=user)
+        if status == 200:
+            self.assertEqual(membership['count'], 1)
+            self.assertEqual(membership['results'][0]['user']['id'], self.users['student_enrolled'].username)
+
+    @ddt.data(
+        (None, 403, False),
+        ('student_inactive', 403, False),
+        ('student_unenrolled', 200, False),
```

Comment:
`student_unenrolled` receives a valid response with no results. My thinking was that returning anything else could leak the existence of usernames.